src/widgets/components/content.rs

```diff
@@ -185,6 +223,29 @@ impl AsyncComponent for ContentComponentModel {
 		_root: &Self::Root,
 	) {
 		match message {
+			ContentComponentInput::RevealTaskDetails(index, task) => {
+				self.show_task_details = true;
+				if self.selected_task.is_none()
+					|| self.selected_task.as_ref().unwrap().id != task.id
+				{
+					self.selected_task = Some(task.clone());
+					self.task_details_factory.guard().clear();
+					self
+						.task_details_factory
+						.guard()
+						.push_back(TaskDetailsFactoryInit::new(task, index));
```

Comment:
It's nicer to get the guard only once if you're doing multiple operations. Every time the guard is dropped, it updates the widgets of the factory and less updates are always more efficient.

```rust
let mut guard = self.task_detail_factory.guard();
guard.clear();
guard.push_bask(...);
```